### PR TITLE
fix(message-input): image picker

### PR DESCRIPTION
**Pull request by sanjeevdialpad**
# Image picker in message input

## :hammer_and_wrench: Type Of Change

- [x] Fix
- [ ] Feature
- [ ] Refactoring
- [ ] Documentation

## :book: Description

changed Dt input to include file type. Used hidden prop to visually hide the input element and also include the click handler on the image button.

## :pencil: Checklist

- [ ] I have reviewed my changes
- [ ] I have added tests
- [ ] I have added all relevant documentation
- [ ] I have validated components with a screen reader
- [ ] I have validated components keyboard navigation
- [ ] I have considered the performance impact of my change
- [ ] I have checked that my change did not significantly increase bundle size
- [ ] I am exporting any new components or constants in the index.js in the component directory
- [ ] I am exporting any new components or constants in the index.js in the root

### CODE REVIEW

File: components/input/input.vue [hunk 1]
```diff
@@ -2,6 +2,7 @@
   <div
     ref="container"
     class="base-input"
+    :class="{ 'd-vi-hidden': hidden }"
```
Comment (braddialpad) [inline components/input/input.vue]: Generally not great practice to have multiple class attributes. Preferable to be like this:

```
:class="['base-input', { 'd-vi-hidden': hidden }]"
```

File: components/input/input.vue [hunk 2]
```diff
@@ -556,6 +565,9 @@ export default {
 
   methods: {
     inputClasses () {
+      if (this.hidden) {
+        return [];
+      }
```
Comment (braddialpad) [inline components/input/input.vue]: Why is this necessary, does it break something to set these classes when hidden? the dom elements would still exist right?

File: recipes/conversation_view/message_input/message_input.vue [hunk 3]
```diff
@@ -32,14 +32,24 @@
...
               <dt-icon
                 name="image"
                 size="300"
               />
             </template>
+            <dt-input
+              ref="messageInputImageUpload"
+              type="file"
+              class="d-ps-absolute"
+              multiple="multiple"
```
Comment (braddialpad) [inline recipes/conversation_view/message_input/message_input.vue]: If I select multiple files it only fires an event for the first one.

**Review by braddialpad — CHANGES_REQUESTED** [review]
cool!

File: components/input/input.vue (hunk 2 above)
Comment (sanjeevdialpad) [reply components/input/input.vue]: yea actually i could remove both the return statements.

File: recipes/conversation_view/message_input/message_input.vue (hunk 3 above)
Comment (sanjeevdialpad) [reply recipes/conversation_view/message_input/message_input.vue]: yea i just noticed, have no clue why, let me check..

File: components/input/input.vue [hunk 4]
```diff
@@ -570,6 +583,9 @@ export default {
     },
 
     inputWrapperClasses () {
+      if (this.hidden) {
```
Comment (sanjeevdialpad) [inline components/input/input.vue:586]: this is needed. otherwise some padding is coming up.

**Comment by sanjeevdialpad** [comment]
Added event to fire for drag and drop of media as well as fixed review comments. For now using the names of files in event payload (to be replaced later with binary)

**Review by braddialpad — APPROVED** [review]
Looks good thanks!
